### PR TITLE
docs: documentation for the experimental etcd feature

### PR DESCRIPTION
**Pull request by bhandras**

### CODE REVIEW

File: docs/etcd.md [hunk 1]
```diff
@@ -0,0 +1,84 @@
+# Experimental etcd support in LND
+
+With the recent introduction of the `kvdb` interface LND can support multiple
+database backends allowing experimentation with the storage model as well as
+improving robustness trough eg. replicating essential data.
```
Comment (cfromknecht) [inline docs/etcd.md:5]:
```suggestion
improving robustness through eg. replicating essential data.
```

File: docs/etcd.md [hunk 2]
```diff
@@ -0,0 +1,84 @@
...
+has *many* rough edges for the time being. It is therefore highly recommended to
+not use LND on `etcd` in any kind of production environment especially not
+on bitcoin mainnet.
+
+## Building LND with etcd support
+
+To create a dev build of LND with etcd support use the following command:
+
+```
+make tags="autopilotrpc signrpc walletrpc chainrpc invoicesrpc watchtowerrpc kvdb_etcd"
```
Comment (cfromknecht) [inline docs/etcd.md:18]: the other build tags seem irrelevant here?

File: docs/etcd.md [hunk 3]
```diff
@@ -0,0 +1,84 @@
...
+The large `max-txn-ops` and `max-request-bytes` values are currently required in
+case of running LND with the full graph in etcd. Upcoming versions will split
+the database to local and replicated parts and only essential parts will remain
+in the replicated database, removing the requirement for these additional 
+settings.
+
+## Configuring LND to run on etcd
+
+To run LND with etcd, additional configuration is needed, specified either
+trough command line flags or in `lnd.conf`.
```
Comment (cfromknecht) [inline docs/etcd.md:50]:
```suggestion
through command line flags or in `lnd.conf`.
```

File: docs/etcd.md [hunk 4]
```diff
@@ -0,0 +1,84 @@
...
+Optionally users can specifiy `db.etcd.user` and `db.etcd.pass` for db user
+authentication.
+
+## Migrating existing channel.db to etcd
+
+This is currently not supported.
+
+## Disclaimer
+
+As mentined before this is an experimental feature, and with that your data
```
Comment (cfromknecht) [inline docs/etcd.md:83]:
```suggestion
As mentioned before this is an experimental feature, and with that your data
```

File: docs/etcd.md (hunk 3 above)
Comment (bhandras) [reply docs/etcd.md:50]: done

File: docs/etcd.md (hunk 2 above)
Comment (bhandras) [reply docs/etcd.md]: agreed, removed

File: docs/etcd.md (hunk 1 above)
Comment (bhandras) [reply docs/etcd.md:5]: thanks, corrected

File: docs/etcd.md (hunk 4 above)
Comment (bhandras) [reply docs/etcd.md]: fixed

File: docs/etcd.md (hunk 1 above)
Comment (Roasbeef) [reply docs/etcd.md:5]: Still see this in my diff (trough).

File: docs/etcd.md [hunk 5]
```diff
@@ -0,0 +1,84 @@
...
+
+Sample command line:
+
+```
+./lnd-debug \
+    --db.backend=etcd \
+    --db.etcd.host=127.0.0.1:2379 \
+    --db.etcd.certfile=/home/user/etcd/bin/default.etcd/fixtures/client/cert.pem \
+    --db.etcd.keyfile=/home/user/etcd/bin/default.etcd/fixtures/client/key.pem \
+    --db.etcd.insecure_skip_verify
```
Comment (Roasbeef) [inline docs/etcd.md:60]: Is this needed even when specifying the certs above? Or Perhaps we're missing the CA-file here?

Comment (bhandras) [reply docs/etcd.md:60]: Yes without `CA` we need `insecure_skip_verify`, otherwise connect will hang LND.

File: docs/etcd.md [hunk 6]
```diff
@@ -0,0 +1,84 @@
...
+
+To start your local etcd instance for testing run:
+
+```
+./etcd \
+    --auto-tls \
+    --advertise-client-urls=https://127.0.0.1:2379 \
+    --listen-client-urls=https://0.0.0.0:2379 \
+    --max-txn-ops=16384 \
+    --max-request-bytes=104857600
```
[the remark this answers is not in the document]
Comment (bhandras) [reply docs/etcd.md:38]: I'll do a run on mainnet to see but for me testnet config worked out well with these values. I'm thinking maybe we should be more directly indicate that running on mainnet is at this point not recommended. WDYT?

File: docs/etcd.md [hunk 7]
```diff
@@ -0,0 +1,84 @@
...
+```
+
+The large `max-txn-ops` and `max-request-bytes` values are currently required in
+case of running LND with the full graph in etcd. Upcoming versions will split
+the database to local and replicated parts and only essential parts will remain
+in the replicated database, removing the requirement for these additional
```
[the remark this answers is not in the document]
Comment (bhandras) [reply docs/etcd.md:44]: For most users having the graph in the local db will help. Are you thinking about users with many invoices? Do you think we should also add comment about why these params needed?

**Comment by Roasbeef** [comment]
Are there any blockers on this?
